experimental/devices/ccs811/ccs811.go

```diff
@@ -0,0 +1,361 @@
+	opts *Opts
+}
+
+const ( //registers
+	statusReg          byte = 0x00
+	measurementModeReg byte = 0x01
+	algoResultsReg     byte = 0x02
+	rawDataReg         byte = 0x03
+	environmentReg     byte = 0x05
+	baselineReg        byte = 0x11
+	resetReg           byte = 0xFF
+)
+
+func (d *Dev) String() string {
+	return "CCS811"
+}
+
+// StartSensorApp initializes sensor to application mode
+func (d *Dev) StartSensorApp() error {
+	err := d.c.Tx([]byte{0xf4}, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// SetMeasurementMode sets one of the 5 measurement modes, interrupt generation
+// and interrupt threshold
+//
+// generateInterrupt:
+//		if true, CCS811 will trigger interrupt when new data is available
+// useThreshold:
+// 		if true, you have to set Threshold register with appropriate values
+func (d *Dev) SetMeasurementMode(measurementMode byte, generateInterrupt, useThreshold bool) error {
+	mesModeValue := (measurementMode << 4)
+	if generateInterrupt {
+		mesModeValue = mesModeValue | (0x1 << 3)
+	}
+	if useThreshold {
+		mesModeValue = mesModeValue | (0x1 << 2)
+	}
+
+	// set measurement mode
+	err := d.c.Tx([]byte{measurementModeReg, mesModeValue}, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type MeasurementMode struct {
+	MeasurementMode   byte
+	GenerateInterrupt bool
+	UseThreshold      bool
+}
+
+func (d *Dev) GetMeasurementMode() (mm MeasurementMode, err error) {
+	r := make([]byte, 1)
+	err = d.c.Tx([]byte{measurementModeReg}, r)
+	if err != nil {
+		return mm, err
+	}
+	mode := r[0] >> 4
+	threshold := (r[0]&4 == 1)
+	interrupt := (r[0]&8 == 1)
+
+	return MeasurementMode{MeasurementMode: mode, GenerateInterrupt: interrupt, UseThreshold: threshold}, nil
+}
+
+// Reset sets device into the BOOT mode
+func (d *Dev) Reset() error {
+	if err := d.c.Tx([]byte{resetReg, 0x11, 0xE5, 0x72, 0x8A}, nil); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ReadStatus returns value of status register
+func (d *Dev) ReadStatus() (byte, error) {
+	r := make([]byte, 1)
+	if err := d.c.Tx([]byte{statusReg}, r); err != nil {
+		return 0, err
+	}
+	return r[0], nil
+}
+
+// ReadRawData provides current and voltage on the sensor
+// current in uA
+// voltage from 0-1023, where 1023 = 1.65V
+func (d *Dev) ReadRawData() (current int, voltage int, err error) {
+	r := make([]byte, 2)
+	if err = d.c.Tx([]byte{rawDataReg}, r); err != nil {
+		return 0, 0, err
+	}
+	current, voltage = valuesFromRawData(r)
+	return
+}
+
+// SetEnvironmentData allows to provide temperature and humidity so
+// sensor can compensate it's measurement
+func (d *Dev) SetEnvironmentData(temp, humidity float32) error {
+	rawTemp := uint16((temp + 25) * 512)
+	rawHum := uint16(humidity * 512)
+	w := []byte{environmentReg,
+		byte(rawHum >> 8),
+		byte(rawHum),
+		byte(rawTemp >> 8),
+		byte(rawTemp)}
+
+	if err := d.c.Tx(w, nil); err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetBaseline provides current baseline used by measurement alogrithm
```

Comment:
"algorithm"
but really, that doesn't tell the user much. :) I'd recommend to say "See SetBaseline for the choice of algorithms" or something like that.

Reply:
It's related to internal algorihtm of changing voltage and current of sensor to VOC/eCO2. User doesn't have chance to influence it. I'll try to be more clear.